lib/linguist/heuristics.yml

```diff
@@ -675,7 +675,7 @@ disambiguations:
 - extensions: ['.stl']
   rules:
   - language: STL
-    pattern: '\A\s*solid(?=$|\s)(?m:.*?)\Rendsolid(?:$|\s)'
+    pattern: '\A\s*solid(?=$|\s)(?:.|[\r\n])*?^endsolid(?:$|\s)'
```

Comment:
Oooof!! This new regex is causing timeouts on GitHub.com. Why? Because it suffers from catastrophic backtracking with larger files... precisely where this change is as can be seen at https://regex101.com/r/2JKkxf/1 (contrived example using the [sample we have](https://github.com/github-linguist/linguist/blob/cc87d62470781ae3d0875f4478e85c8c204fcf96/test/fixtures/Generic/stl/STL/cube1.stl?short_path=c542311))

The old regex doesn't hit this problem as it doesn't seem to be matching the same thing and determines this much sooner 😁 - https://regex101.com/r/gBFDDP/1

/cc @Alhadis @jorendorff

Reply:
The reason the old regex doesn't match anything in regex101 is that regex101 is using PCRE, not Ruby, and the old regex uses Ruby-specific features that PCRE doesn't understand.

Reply:
I tried both the old regex and the new one on [a 5000-line file](https://github.com/prusa3d/PrusaSlicer/blob/c05542590d7c2d73eb69bbf7a82a482a075815c1/resources/profiles/Sovol/SV05_bed.stl).

I used this Ruby script.

```ruby
require 'benchmark'

LAST_GOOD_RE = /\A\s*solid(?=$|\s)(?m:.*?)\Rendsolid(?:$|\s)/
FIRST_BAD_RE = /\A\s*solid(?=$|\s)(?:.|[\r\n])*?^endsolid(?:$|\s)/

text = File.read("SV05_bed.stl")

t = Benchmark.measure {
  LAST_GOOD_RE.match?(text)
}
puts t.real

t = Benchmark.measure {
  FIRST_BAD_RE.match?(text)
}
puts t.real
```

If I'm reading this right, the old regex runs in 2ms and the new one runs in 4ms, on my laptop. So it really is slower.